service/lexruntimeservice/endpoints_test.go

```diff
@@ -0,0 +1,444 @@
+/*
+# Considerations
+* Could we eliminate pseudo-regions from the SDK?
+    * Do we need to support backwards compatibility with pseudo-regions and "clean" them if read in from ~/.aws/config
+    * Backwards compatibility would effectively duplicate logic between Java endpoint codegen and Go
+* Still need to support -global variants
+* Could the resolver be configurable with grab bag attributes to enable subset of features
+	* FIPS
+	* DualStack
+	* S3 Accelerate
+* What is the desired behavior when a region is not modeled but endpoint attributes are specified?
+    * Should these fail? Should we attempt to define a non-ratified service URL format?
+* Java V2 just has logic in it's region validation to allow matching of regions that either have fips- or -fips prefix
+or suffix when validating with a regex by effectively stream those components off
+*/
+
+package lexruntimeservice
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+var serviceName string
+
+type EndpointResolver interface {
+	ResolveEndpoint(region string, options ...func(*ResolverOptions)) (aws.Endpoint, error)
+}
+
+func NewDefaultResolver(options ...func(options *ResolverOptions)) AWSEndpointProvider {
+	opts := &ResolverOptions{}
+
+	for _, fn := range options {
+		fn(opts)
+	}
+
+	return AWSEndpointProvider{
+		options:    *opts,
+		partitions: defaultPartitions,
+	}
+}
+
+type ResolverOptions struct {
+	Constraints  *Capabilities
+	DisableHTTPS bool
+}
+
+type AWSEndpointProvider struct {
+	options    ResolverOptions
+	partitions partitions
+}
+
+func (d AWSEndpointProvider) ResolveEndpoint(region string, options ...func(o *ResolverOptions)) (aws.Endpoint, error) {
+	ro := d.options
+	for _, fn := range options {
+		fn(&ro)
+	}
+	return d.partitions.ResolveEndpoint(region, ro)
+}
+
+var defaultPartitions partitions
+
+func init() {
+	defaultPartitions = partitions{
+		partition{
+			ID: "aws",
+			RegionRegex: func() *regexp.Regexp {
+				reg, _ := regexp.Compile("^(us|eu|ap|sa|ca|me)\\-\\w+\\-\\d+$")
+				return reg
+			}(),
+			Defaults: []endpoint{
+				// Default from endpoints.json
+				{
+					Hostname:    "runtime.lex.{region}.amazonaws.com",
+					Protocols:   []string{"https"},
+					SigningName: "lex",
+				},
+				// Optional inferred template with capabilities
+				{
+					Hostname:  "runtime-fips.lex.{region}.amazonaws.com",
+					Protocols: []string{"https"},
+					InferredCapabilities: &Capabilities{
+						FIPS: true,
+					},
+					SigningName: "lex",
+				},
+			},
+			Endpoints: map[string]endpoint{
+				"us-west-2": {
+					Hostname:  "runtime.lex.us-west-2.amazonaws.com",
+					Protocols: []string{"https"},
+				},
+				"us-west-2-fips": {
+					Hostname:  "runtime-fips.lex.us-west-2.amazonaws.com",
+					Protocols: []string{"https"},
+					InferredCapabilities: &Capabilities{
+						FIPS: true,
+					},
```

Comment:
nit should this be a pointer since all of the other types are values?

Reply:
this looks like old artifact history